core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java

```diff
@@ -478,4 +482,18 @@ public FlowIdentifier modifyProxyConfig(NameOrCrn nameOrCrn, String accountId, S
         MDCBuilder.buildMdcContext(stack);
         return flowManager.triggerModifyProxyConfig(stack.getId(), previousProxyConfigCrn);
     }
+
+    public FlowIdentifier triggerSecretRotation(String crn, List<String> secrets, RotationFlowExecutionType executionType) {
+        StackView viewByCrn = stackDtoService.getStackViewByCrn(crn);
+        Long stackId = viewByCrn.getId();
+        List<SecretType> secretTypes = secrets.stream()
```

Comment:
Can you please make sure that even if a secret type occurs multiple times in the list we still rotate it once or throw an error. Probably throwing an exception explicitly is better so the user can be sure about the order.

Reply:
I think we cannot enforce any order from customer, thus I would only remove duplications, if there is any. If we need to be stricter, we can change that later.

Reply:
discussed, fixed, added UT for it